python/core/unified_api_server.py

```
@@ -987,68 +1296,401 @@ async def get_comprehensive_analytics(
+                    'accuracy': (analysis or game).get('accuracy'),
+                    'time_management_score': (analysis or game).get('time_management_score'),
+                    'blunders': (analysis or game).get('blunders'),
+                    'opponent_blunders': (analysis or game).get('opponent_blunders')
+                })
+
+            # Resignation timing derived from PGN header
+            termination = _parse_termination_from_pgn(pgn_map.get(game_id)) if pgn_map else None
+            if termination:
+                # Check for resignations
+                if 'resign' in termination.lower():
+                    if 'opponent' in termination.lower() or 'resigned' in termination.lower() and result == 'win':
```

Ruff (0.14.1)

Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)